vm/src/stdlib/array.rs

```diff
@@ -1037,6 +1059,47 @@ mod array {
             }
             Ok(true)
         }
+
+        #[pymethod(magic)]
+        fn reduce_ex(
+            zelf: PyRef<Self>,
+            proto: usize,
+            vm: &VirtualMachine,
+        ) -> (PyObjectRef, PyTupleRef, Option<PyDictRef>) {
+            let array = zelf.read();
+            let cls = zelf.as_object().clone_class().into_object();
+            let typecode = vm.ctx.new_str(array.typecode_str());
+            if proto < 3 {
+                let values = vm.ctx.new_list(array.get_objects(vm));
+                (
+                    cls,
+                    PyTupleRef::with_elements(vec![typecode, values], &vm.ctx),
+                    zelf.as_object().dict(),
+                )
+            } else {
+                let bytes = vm.ctx.new_bytes(array.get_bytes().to_vec());
+                let code = MachineFormatCode::from_typecode(array.typecode()).unwrap();
+                let code = PyInt::from(u8::from(code)).into_object(vm);
+                let func = _array_reconstructor.into_func();
+                let func =
+                    PyNativeFuncDef::new(func, PyStr::from("_array_reconstructor").into_ref(vm));
+                let module = zelf.class().module(vm);
+                let func = func.into_function().with_module(module).build(&vm.ctx);
+                (
+                    func,
```

Comment:
How could I get the module function _array_reconstructor here?

Reply:
`vm.import` followed by a `get_attribute` should also work.

Reply:
I wish we can access them through VirtualMachine in future.